src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala

```diff
@@ -1061,11 +1055,38 @@ object ModelFactory {
   val defaultGroupDesc = None
   val defaultGroupPriority = 1000
 
-  val tokens = raw"€\{($FILE_PATH|$FILE_EXT|$FILE_PATH_EXT|$FILE_LINE|$TPL_OWNER|$TPL_NAME)\}".r
+  // 4 subgroups: (.?), (FILE_PATH|FILE_PATH_EXT), (FILE_EXT) and (FILE_LINE|TPL_OWNER|TPL_NAME). non-matching are `null`
+  val tokens = raw"(?:(.?)€\{($FILE_PATH|$FILE_PATH_EXT)\}|\.?€\{($FILE_EXT)\}|€\{($FILE_LINE|$TPL_OWNER|$TPL_NAME)\})".r
   final val FILE_PATH     = "FILE_PATH"
   final val FILE_EXT      = "FILE_EXT"
   final val FILE_PATH_EXT = "FILE_PATH_EXT"
   final val FILE_LINE     = "FILE_LINE"
   final val TPL_OWNER     = "TPL_OWNER"
   final val TPL_NAME      = "TPL_NAME"
+
+  val WordChar = raw"(\w)".r
+
+  def expandUrl(urlTemplate: String, filePath: String, fileExt: String, filePathExt: String, line: Int, tplOwner: String, tplName: String): String = {
+    val absolute = filePath.startsWith("/")
+
+    def subst(groups: List[String]): String = {
+      // If a relative path follows a word character, insert a `/`
+      def sep: String = groups.head match {
+        case null => ""
+        case WordChar(c) if !absolute => c + "/"
+        case c => c
+      }
+
+      groups.tail.find(_ != null).get match {
+        case FILE_PATH => s"$sep$filePath"
+        case FILE_EXT => fileExt
+        case FILE_PATH_EXT => s"$sep$filePathExt"
+        case FILE_LINE => line.toString
+        case TPL_OWNER => tplOwner
+        case TPL_NAME => tplName
```

Comment:
remember when arrows used to line up?

Reply:
There are other things that itch me, but somehow not this...